### PR TITLE
added pubsub functionality

### PR DESCRIPTION
**Pull request by AbhayMaheshwari123**

### CODE REVIEW

File: src/components/NumberGenerator/NumberGenerator.tsx [hunk 1]
```diff
@@ -2,6 +2,9 @@ import React, { useCallback, useState } from "react";
 import { random } from "lodash";
 import { Button } from "@mui/material";
 import { START_VALUE, END_VALUE } from "../../util/config";
+import {NumberGeneratorEvent} from "../../util/event/NumberGeneratorEvent";
```
Comment (nik-1207) [inline src/components/NumberGenerator/NumberGenerator.tsx]: replace it with custom onclick event

File: src/util/event/NumberGeneratorEvent.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export class NumberGeneratorEvent{
+    args: number;
```
Comment (nik-1207) [inline src/util/event/NumberGeneratorEvent.ts]: replace args with some other name

File: src/util/pubsub/PubSub.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export function publish(event:{args:number}){
```
Comment (nik-1207) [inline src/util/pubsub/PubSub.ts]: replace args with other name

**Review by nik-1207 — CHANGES_REQUESTED** [review]

File: src/components/NumberGenerator/NumberGenerator.tsx (hunk 1 above)
Comment (AbhayMaheshwari123) [reply src/components/NumberGenerator/NumberGenerator.tsx]: resolved

File: src/util/event/NumberGeneratorEvent.ts (hunk 2 above)
Comment (AbhayMaheshwari123) [reply src/util/event/NumberGeneratorEvent.ts]: should be args, so it can be understand well that where and what are arguments

**Pull request merged**